ledger/nomos-ledger/src/mantle/sdp/mod.rs

```diff
@@ -220,10 +226,50 @@ impl SdpLedger {
         }
     }
 
-    // TODO: genesis init
-    #[cfg(test)]
+    pub fn from_genesis_tx(
+        config: &Config,
+        utxo_tree: &UtxoTree,
+        tx: impl GenesisTx,
+    ) -> Result<Self, Error> {
+        let mut sdp = Self::new()
+            .with_service(ServiceType::BlendNetwork)
+            .with_service(ServiceType::DataAvailability);
+
+        let ops = tx.mantle_tx().ops.iter().zip(tx.op_proofs().iter());
+        for (op, proof) in ops {
+            if let (
+                Op::SDPDeclare(op),
+                Some(OpProof::ZkAndEd25519Sigs {
+                    zk_sig,
+                    ed25519_sig,
+                }),
+            ) = (op, proof)
+            {
+                let Some((utxo, _)) = utxo_tree.utxos().get(&op.locked_note_id) else {
+                    return Err(Error::NoteNotFound(op.locked_note_id));
+                };
+                sdp =
+                    sdp.apply_declare_msg(op, utxo.note, zk_sig, ed25519_sig, tx.hash(), config)?;
+            }
+        }
```

Comment:
ok I see what you meant now, we already have a `from_genesis_tx` in mantle ledger that does most of these operations anyway (e.g checking invalid ops and fetching notes). What aobut we have `from_genesis` with a list of declarations / proofs instead? Otherwise this is good